### DISCUSSION TITLE
How ds resolve the  Brain-Split and Network-Partition with zk

### DISCUSSION THREAD

**Discussion by krystalics** [apache/dolphinscheduler#7797] [Q&A]
I wanna ask the point to resolve the  Brain-Split and Network-Partition problem in ds with zk. 
does some document can i scan

**Comment by itschenxiang** [comment]
in version: 3.2.1, I thought you can get part of the answer from the offical documentation:
https://dolphinscheduler.apache.org/zh-cn/docs/3.2.1/architecture/design
![image](https://github.com/apache/dolphinscheduler/assets/42402297/9bbc974e-43c8-4377-a84a-d2202b24a595)

post in en:
https://medium.com/@ApacheDolphinScheduler/the-key-points-in-designing-the-masterserver-of-distributed-visual-job-scheduling-platform-c8ce70a25b58
![image](https://github.com/apache/dolphinscheduler/assets/42402297/fab21d5a-5bdf-42b3-b430-16917bed12cb)



from the source code, you can find the answer from there (failover by zk):
https://github.com/apache/dolphinscheduler/blob/3.2.1-release/dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-zookeeper/src/main/java/org/apache/dolphinscheduler/plugin/registry/zookeeper/ZookeeperConnectionStateListener.java#L38
